test/performance/commCompare/gbt/many-to-one.chpl

```diff
@@ -0,0 +1,98 @@
+use Time;
+use CommDiagnostics;
+
+const numWorkerNodes = numLocales - 1;
+config const numTasksPerNode = min reduce ([i in 1..numWorkerNodes]
+                                           Locales(i).maxTaskPar);
+
+config const numOps = 10**6;
+const numOpsPerTask = numOps / (numWorkerNodes * numTasksPerNode);
+
+config const printConfig = false;
+config const printTimings = false;
+config const printCommDiags = false;
+
+var x0: int;
+var x0Atomic: atomic int;
+
+enum op_t {
+  opNone,
+  opGet,
+  opPut,
+  opAMO,
+  opFastOn,
+  opOn
+};
+
+use op_t;
+
+config param op = opNone;
+
+proc main() {
+  if printConfig {
+    writeln(numWorkerNodes, ' worker nodes, ',
+            numTasksPerNode, ' worker tasks per node, ',
+            numOps, ' ', op, (if numOps == 1 then '' else 's'));
+  }
+
+  if printCommDiags {
+    resetCommDiagnostics();
+    startCommDiagnostics();
+  }
+
+  const startTime = getCurrentTime();
+
+  coforall locIdx in 1..numWorkerNodes with ( ref x0, ref x0Atomic ) {
+    on Locales(locIdx) {
+      coforall 1..numTasksPerNode with ( ref x0, ref x0Atomic ) {
+        for i in 1..numOpsPerTask {
+          select op {
+            when opNone {
+            }
+
+            when opGet {
+              if i == 1 then x0 = 0; // prevent compiler loop-hoisting GET
```

Comment:
Do you actually see this happening? If so that's a LICM bug I need to fix

Reply:
I did see it happening, but is it really a LICM bug?  I figured LICM didn't see any stores within the loop, so it assumed that in the absence of ordering constraints, it could hoist the load out of the loop.

Reply:
Ah, I guess you're right. Since the extern proc is declared without an intent, it should default to in-intent, in which case we can assume `x0` isn't changing. I think if you declared it with a ref or an inout intent we wouldn't hoist (I actually don't know if it's legal to do mark extern procs with inout)

Reply:
If it was `ref` we wouldn't even do the GET, because we'd pass the wide pointer rather than the value as the actual argument.  I don't think `inout` would work either, because then we'd probably do a GET and then pass the pointer and then do a PUT afterward.